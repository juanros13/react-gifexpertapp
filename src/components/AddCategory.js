import React, { useState } from 'react';
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');

    const hanledInputChange = (e) =>{
        setInputValue(e.target.value);
        // console.log(e.target.value);
    }
    const hanledSubmit = (e) => {
        e.preventDefault();
        // console.log('submit hecho', inputValue);
        if(inputValue.trim().length>2){
            setCategories(cats => [ inputValue,...cats,]);
            setInputValue('');
        }
        
    }

    return (
        <form onSubmit={ hanledSubmit }>
            {/* <h1>{ inputValue }</h1> */}
            <p> { inputValue } </p>
            <input
                type='text'
                value={ inputValue }
                onChange={ hanledInputChange }
                
            />
        </form>
    )
}
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
