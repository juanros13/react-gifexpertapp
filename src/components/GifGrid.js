import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
// import { getGifs } from '../helpers/getGifs';
import GifGridItem  from './GifGridItem';

import PropTypes from 'prop-types';

export const GifGrid = ( { category } ) => {
    // const [counter, setCounter] = useState(0);
    const { data:images, loading } = useFetchGifs(category);

    // const [images, setImages] = useState([])
    // useEffect(()=>{
    //     getGifs(category)
    //     .then(imgs => setImages(imgs));
    // },[category]);

    //getGifs();
    return (
        <>
        <h3 className='animate__animated animate__fadeIn'>{ category }</h3>
        { loading && <p className='animate__animated animate__flash'>Loading</p>}

        <div className='card-gif'>
            {
                images.map((img) =>
                    // <li  key = { id }>{ title }</li>
                    <GifGridItem
                        key = { img.id }
                        { ...img }
                    />
                )
            }
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}