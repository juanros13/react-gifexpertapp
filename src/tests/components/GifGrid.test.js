import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import '@testing-library/jest-dom';
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Pruenas para el componente <GrifGid />', () => {
    const categoria = 'One punch';
    test('Debe mostrarse correctamente el GrifGrid', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category= { categoria } />);
        
        expect(wrapper).toMatchSnapshot();
        
    });
    test('debe mostrar items cuando se cargan imnaegens con useFetchGifs ', () => {

        const gifs = [
            {
                id: 'abc',
                title: 'Cualquier cosa',
                url: 'http://lcalhost/loquesea.jpg'
            },
            {
                id: 'abc',
                title: 'Cualquier cosa',
                url: 'http://lcalhost/loquesea.jpg'
            }
        ];
        useFetchGifs.mockReturnValue({
          data : gifs,
          loading: false
        });
        const wrapper = shallow(<GifGrid category= { categoria } />);

        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe(gifs.length)



    });
    
    
})
