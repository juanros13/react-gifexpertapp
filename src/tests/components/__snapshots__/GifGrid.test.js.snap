// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruenas para el componente <GrifGid /> Debe mostrarse correctamente el GrifGrid 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-gif"
  />
</Fragment>
`;

exports[`Pruenas para el componente <GrifGid /> debe mostrar items cuando se cargan imnaegens con useFetchGifs  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One punch
  </h3>
  <div
    className="card-gif"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Cualquier cosa"
      url="http://lcalhost/loquesea.jpg"
    />
    <GifGridItem
      id="abc"
      key="abc"
      title="Cualquier cosa"
      url="http://lcalhost/loquesea.jpg"
    />
  </div>
</Fragment>
`;
